Test/Models/ConcatenatedModel.cpp

```diff
@@ -43,49 +44,58 @@ BOOST_AUTO_TEST_CASE( CONCATENATED_MODEL_Value )
 		net2Params(i)=shark::random::uni( shark::random::globalRng,-1,1);
 		modelParams(i+net1Params.size())=net2Params(i);
 	}
+	RealVector net3Params(net3.numberOfParameters());
+	for(size_t i=0;i!=net3Params.size();++i){
+		net3Params(i)=shark::random::uni( shark::random::globalRng,-1,1);
+		modelParams(i+net1Params.size() + net2Params.size())=net3Params(i);
+	}
 	//check whether parameter copying is working
 	model.setParameterVector(modelParams);
 	double error1=norm_sqr(net1Params-net1.parameterVector());
 	double error2=norm_sqr(net2Params-net2.parameterVector());
-	double error3=norm_sqr(modelParams-model.parameterVector());
+	double error3=norm_sqr(net3Params-net3.parameterVector());
+	double error4=norm_sqr(modelParams-model.parameterVector());
 	BOOST_CHECK_EQUAL(error1,0.0);
 	BOOST_CHECK_EQUAL(error2,0.0);
 	BOOST_CHECK_EQUAL(error3,0.0);
+	BOOST_CHECK_EQUAL(error4,0.0);
 
 	//test Results;
 	RealVector input(3);
 	for(size_t i=0;i!=3;++i){
 		input(i)=shark::random::uni( shark::random::globalRng,-1,1);
 	}
-	RealVector intermediateResult = net1(input);
-	RealVector endResult = net2(intermediateResult);
+	RealVector intermediateResult1 = net1(input);
+	RealVector intermediateResult2 = net2(intermediateResult1);
+	RealVector endResult = net3(intermediateResult2);
 
 	//evaluate point
 	RealVector modelResult = model(input);
 	double modelError = norm_sqr(modelResult-endResult);
-	BOOST_CHECK_SMALL(modelError,1.e-35);
+	BOOST_CHECK_SMALL(modelError,1.e-14);
 }
 BOOST_AUTO_TEST_CASE( CONCATENATED_MODEL_weightedParameterDerivative )
 {
-	FFNet<LogisticNeuron,LogisticNeuron> net1;
-	LinearModel<> net2;
-	net1.setStructure(3,5,2);
-	net2.setStructure(2,4);
-	ConcatenatedModel<RealVector,RealVector> model (&net1,&net2);
+	LinearModel<> net1;
+	NeuronLayer<SoftmaxNeuron<> > net2(5);
+	LinearModel<> net3;
+	net1.setStructure(3,5,1);
+	net3.setStructure(5,2,1);
+	ConcatenatedModel<RealVector> model  = net1 >> net2 >> net3;
 
 	//test1: all activated
 	{
 		//parameters
-		size_t modelParameters = net1.numberOfParameters()+net2.numberOfParameters();
+		size_t modelParameters = net1.numberOfParameters()+net3.numberOfParameters();
```

Comment:
If this case is "all activated" as the comment states, then we should also add the number of parameters of net2.